universal-application-tool-0.0.1/app/services/program/ProgramService.java

```diff
@@ -38,19 +39,23 @@
    * @param blockName a name for the block to add
    * @param blockDescription a description of what the questions in the block address
    * @return the updated {@link ProgramDefinition}
+   * @throws ProgramNotFoundException when programId does not correspond to a real Program.
    */
-  ProgramDefinition addBlockToProgram(long programId, String blockName, String blockDescription);
+  ProgramDefinition addBlockToProgram(long programId, String blockName, String blockDescription)
+      throws ProgramNotFoundException;
 
   /**
    * Update a {@link BlockDefinition} with a set of questions.
    *
    * @param programId the ID of the program to update
    * @param blockDefinitionId the ID of the block to update
-   * @param questionDefinitionIds an {@link ImmutableList} of questions for the block
+   * @param questionDefinitions an {@link ImmutableList} of questions for the block
    * @return the updated {@link ProgramDefinition}
+   * @throws ProgramNotFoundException when programId does not correspond to a real Program.
    */
   ProgramDefinition setBlockQuestions(
```

Comment:
suggestion (non-blocking): I think eventually we'll want to use the ErrorAnd class here, so we can return validation errors (for example, if there is no block with this ID for this program)

Reply:
ahh good call!

Reply:
Let's make this a separate PR.